servers/audio/audio_stream.cpp

```diff
@@ -322,26 +322,25 @@ AudioStreamMicrophone::AudioStreamMicrophone() {
 }
 
 int AudioStreamPlaybackMicrophone::_mix_internal(AudioFrame *p_buffer, int p_frames) {
-	AudioDriver::get_singleton()->lock();
-
-	Vector<int32_t> buf = AudioDriver::get_singleton()->get_input_buffer();
-	unsigned int input_size = AudioDriver::get_singleton()->get_input_size();
+	const Vector<int32_t> &buf = AudioDriver::get_singleton()->get_input_buffer();
+	AudioDriver::SizePosition size_position = AudioDriver::get_singleton()->get_input_size_position();
 	int mix_rate = AudioDriver::get_singleton()->get_mix_rate();
 	unsigned int playback_delay = MIN(((50 * mix_rate) / 1000) * 2, buf.size() >> 1);
-#ifdef DEBUG_ENABLED
-	unsigned int input_position = AudioDriver::get_singleton()->get_input_position();
-#endif
 
 	int mixed_frames = p_frames;
 
-	if (playback_delay > input_size) {
+	if (playback_delay > size_position.size) {
 		for (int i = 0; i < p_frames; i++) {
 			p_buffer[i] = AudioFrame(0.0f, 0.0f);
 		}
 		input_ofs = 0;
 	} else {
-		for (int i = 0; i < p_frames; i++) {
-			if (input_size > input_ofs && (int)input_ofs < buf.size()) {
+		bool was_locked = false;
+
+		int current_frame = 0;
+		for (; current_frame < p_frames; current_frame++) {
+			if (size_position.size > input_ofs && (int)input_ofs < buf.size() &&
+					(input_ofs < size_position.position || (int)(input_ofs - size_position.position) > (p_frames * 2))) {
```

Comment:
Not really sure about `(int)(input_ofs - size_position.position) > (p_frames * 2))`, why exactly this check? This was moved from here:
https://github.com/godotengine/godot/blob/4223444d87e125534c72d09c4f45ed490b0c5e5e/servers/audio/audio_stream.cpp#L365-L367